### PR TITLE
Temporarily fixed Sym's create_project endpoint

### PR DESCRIPTION
**Pull request by peterghrong**
# Summary

1. Fixed Sym's create_project POST endpoint 
2. Added unittest to ensure that the endpoint works correctly.

## Need some input
1. Sym added a check to ensure that all the fields are filled in for the project model before it is able to be posted. But is this really necessary ? @celinehuang

## Test Plan
1. Tested that the unique uuids are generated
2. Tested that the endpoint is used correctly
Screen Shot 2021-01-15 at 11 13 37 AM

## Related Issues
Closes #70

### CODE REVIEW

File: backend/app/project/routes.py [hunk 1]
```diff
@@ -1,8 +1,64 @@
...
 from app import db
-from app.models import Project
 from . import project
+from app.models import Project
+import uuid
+
+
+# get all endpoints
+@project.route("", methods=["GET"])
+def get_projects():
```
Comment (CurtisMIT) [inline backend/app/project/routes.py:10]: @peterghrong Isn't this done in another pr? If so, is there a need to write it here (aka is there a need to merge #98)?

File: backend/tests/test_project.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def tearDown(self):
         db.drop_all()
         self.app_context.pop()
 
-    def test_project_routes(self):
+    def test_update_project(self):
```
Comment (CurtisMIT) [inline backend/tests/test_project.py:22]: @peterghrong same for this.

File: backend/tests/test_project.py [hunk 3]
```diff
@@ -71,3 +71,90 @@ def test_project_routes(self):
             data=json.dumps(update_obj),
         )
         self.assertEqual(response.status_code, 404)
+
+    # test get_projects endpoint
+    def test_get_project(self):
```
Comment (CurtisMIT) [inline backend/tests/test_project.py:76]: @peterghrong and this.

File: backend/tests/test_project.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import json
 import unittest
 import uuid
-from flask import current_app
+from flask import current_app, request
```
Comment (CurtisMIT) [inline backend/tests/test_project.py:4]: @peterghrong Do you have an auto-import type of thing lol? This is unused I think.

File: backend/app/project/routes.py (hunk 1 above)
Comment (peterghrong) [reply backend/app/project/routes.py:10]: Yes I just pulled from the other branch so everything in this branch is somewhat up to date. Apologies if this is bad practice !

File: backend/app/project/routes.py [hunk 5]
```diff
@@ -1,8 +1,64 @@
-import uuid
-from flask import jsonify, request, abort
+from flask import Flask, jsonify, request, abort
```
[the remark this answers is not in the document]
Comment (peterghrong) [reply backend/app/project/routes.py:1]: You right

File: backend/tests/test_project.py (hunk 4 above)
Comment (peterghrong) [reply backend/tests/test_project.py:4]: Honestly possibly at this point with all these dumb errors

File: backend/app/project/routes.py [hunk 6]
```diff
@@ -1,8 +1,60 @@
...
+    new_project = Project(
+        address=address,
+        city=city,
+        province=province,
+        postal_code=postal_code,
+        neighbourhood=neighbourhood,
+        year=year,
+        name=name,
+        type=type,
+        contacts=contacts,
```
Comment (CurtisMIT) [inline backend/app/project/routes.py:51]: @peterghrong Sorry can't believe I missed this, but when we're grabbing contacts from the request payload, would we not want to initialize it into a Contact class before assigning it to the Project? I think this will lead to a runtime error.

File: backend/tests/test_project.py (hunk 4 above)
Comment (CurtisMIT) [inline backend/tests/test_project.py:4]: @peterghrong Is there a reason why this keeps on popping up?

**Review by CurtisMIT — COMMENTED** [review]
@peterghrong Sorry missed a critical part, also is there a reason why the request import keeps on popping up?

File: backend/tests/test_project.py (hunk 4 above)
Comment (peterghrong) [reply backend/tests/test_project.py:4]: Might have pulled from a bad branch lemme just clean everything up

File: backend/app/project/routes.py (hunk 6 above)
Comment (peterghrong) [reply backend/app/project/routes.py:51]: Good point, I think this will depend on the contact endpoints though

File: backend/app/project/routes.py [hunk 7]
```diff
@@ -1,8 +1,61 @@
-import uuid
 from flask import jsonify, request, abort
 from app import db
-from app.models import Project
 from . import project
+from app.models import Project
+
+
+# get all endpoints
+@project.route("", methods=["GET"])
```
Comment (celinehuang) [inline backend/app/project/routes.py:8]: This is the same method as that's created in #98 ?

File: backend/tests/test_project.py (hunk 2 above)
Comment (celinehuang) [inline backend/tests/test_project.py:22]: Tbh next time you could just make a separate PR for this change, in practice, each PR should be scoped to it's own problem so adding random changes here and there makes it harder for reviewers to follow

Comment (peterghrong) [reply backend/tests/test_project.py:22]: Will do

File: backend/app/project/routes.py (hunk 7 above)
Comment (peterghrong) [reply backend/app/project/routes.py:8]: Yes sorry shouldnt have pulled from the #98 branch

File: backend/app/project/routes.py [hunk 8]
```diff
@@ -1,8 +1,57 @@
...
```
Comment (celinehuang) [inline backend/app/project/routes.py:8]: https://github.com/hack4impact-mcgill/mu-crm-tool/blob/1348062bf13075982f3f4718b858fe76f85a46d1/backend/app/project/routes.py#L16 I think one of these will need to change bc right now both requests would be routed to the same endpoint

File: backend/app/project/routes.py [hunk 9]
```diff
@@ -1,8 +1,57 @@
...
+def get_projects():
+    projects = Project.query.all()
+    return jsonify(Project.serialize_list(projects))
+
+
+# create a new project
+@project.route("", methods=["POST"])
```
Comment (celinehuang) [inline backend/app/project/routes.py:15]:
```suggestion
@project.route("/create-project", methods=["POST"])

Comment (peterghrong) [reply backend/app/project/routes.py:15]: Yes sorry again, should not have pulled from a while back, please ignore this branch for now. I'm hoping that we get the "get_project" endpoint merged in and this branch should be trivial after that.

Comment (peterghrong) [reply backend/app/project/routes.py:8]: I have not changed this yet because according to the articles I have attached below, the best practice is to not expose CRUD function names. Let me know if you decide against this Im more than happy to quickly amend it. 

Refer to "Never use CRUD function names in URIs"
https://stoplight.io/blog/crud-api-design/

The article also raises questions to the update endpoint, as `@project.route("/<uuid:id>/update", methods=["PUT"])` directly exposes the update action.

Comment (CurtisMIT) [reply backend/app/project/routes.py:8]: I'm ok with just `""` as that's partly why I suggested `""` for #98 . @celinehuang If you're ok with it, @peterghrong do you mind changing it for the update route as well?

Comment (peterghrong) [reply backend/app/project/routes.py:8]: Sure thing

**Review by celinehuang — APPROVED** [review]
LGTM
